ui/src/app/components/workflows/workflow-form/workflow-form.component.ts

```diff
@@ -68,13 +69,51 @@ export class WorkflowFormComponent implements OnDestroy {
   workflowSubscription: Subscription;
   confirmationDialogServiceSubscription: Subscription = null;
 
+  initialWorkflowData: {
+    details: WorkflowEntryModel[];
+    sensor: WorkflowEntryModel[];
+    jobs: JobEntryModel[];
+  };
+
   constructor(
     private store: Store<AppState>,
     private confirmationDialogService: ConfirmationDialogService,
     private previousRouteService: PreviousRouteService,
     private router: Router,
   ) {}
 
+  ngOnInit() {
+    this.initialWorkflowData = this.workflowData;
+  }
+
+  hasWorkflowChanged(): boolean {
+    return !(
+      this.areWorkflowEntriesEqual(this.workflowData.details, this.initialWorkflowData.details) &&
+      this.areWorkflowEntriesEqual(this.workflowData.sensor, this.initialWorkflowData.sensor) &&
+      this.areJobsEqual(this.workflowData.jobs, this.initialWorkflowData.jobs)
+    );
+  }
+
+  areWorkflowEntriesEqual(leftEntries: WorkflowEntryModel[], rightEntries: WorkflowEntryModel[]): boolean {
+    return leftEntries
+      .map((leftEntry) =>
+        rightEntries.some((rightEntry) => isEqual(rightEntry.value, leftEntry.value) && rightEntry.property == leftEntry.property),
+      )
+      .every((entry) => entry);
+  }
+
+  areJobsEqual(leftJobEntry: JobEntryModel[], rightJobEntry: JobEntryModel[]): boolean {
```

Comment:
Could you please write at least four tests for this method ? It should be easy.
One test when they are equal
one when they are not (difference in properties and length is the same)
one when they are not (difference in length)
one when they are empty.